src/lib/profiles/data-management/Current/SubscriptionClient.cpp

```diff
@@ -551,8 +546,11 @@ WEAVE_ERROR SubscriptionClient::SendSubscribeRequest(void)
                         outSubscribeParam.mSubscribeRequestPrepareNeeded.mPathList[i].mPropertyPathHandle;
                 }
 
-                err = mDataSinkCatalog->Locate(versionedTraitPath.mTraitDataHandle, &pDataSink);
-                SuccessOrExit(err);
+                if (mDataSinkCatalog->Locate(versionedTraitPath.mTraitDataHandle, &pDataSink) != WEAVE_NO_ERROR)
+                {
+                    // The application can remove Sinks from the catalog. Ignore this path.
```

Comment:
Nit: Rephrase as 
"Locate() can return an error if the sink has been removed from the catalog. In that case, continue to next entry".